aws/resource_aws_waf_rule.go

```diff
@@ -109,6 +117,23 @@ func resourceAwsWafRuleRead(d *schema.ResourceData, meta interface{}) error {
 		predicates = append(predicates, predicate)
 	}
 
+	arn := arn.ARN{
```

Comment:
Nit: For the future, it might be nice to expose the ARN as an attribute as well if folks need it for other reasons (and we can also use `d.Get("arn").(string)` in the `Update` function as well) 👍 

In the resource code:
```go
// Resource schema
"arn": {
	Type:     schema.TypeString,
	Computed: true,
}

// Read function (or Create function if needed in there first)
arn := arn.ARN{/* ... */}.String()
d.Set("arn", arn)

// Update function can access via
d.Get("arn").(string)
```

Acceptance testing:

```go
// In the _basic test, one of the following
// Depending on global/regional ARN and known resource value (Check) or regex (Match)
testAccCheckResourceAttrGlobalARN(resourceName, "arn", "iam", fmt.Sprintf("policy/%s", rName)),

testAccMatchResourceAttrGlobalARN(resourceName, "arn", "organizations", regexp.MustCompile(`organization/o-.+`)),

testAccCheckResourceAttrRegionalARN(resourceName, "arn", "athena", fmt.Sprintf("workgroup/%s", rName)),

testAccMatchResourceAttrRegionalARN(resourceName, "arn", "acm", regexp.MustCompile(`certificate/.+`)),
```

Resource documentation (attributes section):

```md
* `arn` - Amazon Resource Name (ARN)
```